code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -484,3 +484,30 @@
 	if(prob(15))
 		M.adjustToxLoss(2, 0)
 	..()
+
+/datum/reagent/drug/antidepressant
+	name = "Antidepressant"
+	description = "Feeling down? Take a happy pill!"
+	reagent_state = LIQUID
+	color = "#18a3a3"
+	metabolization_rate = REAGENTS_METABOLISM
+	overdose_threshold = 15
+	addiction_threshold = 10
+
+/datum/reagent/drug/antidepressant/on_mob_metabolize(mob/living/L)
+	var/datum/component/mood/M = L.GetComponent(/datum/component/mood)
+	if(M.mood_level < 5)
+		SEND_SIGNAL(L, COMSIG_ADD_MOOD_EVENT, "antidepressant", /datum/mood_event/antidepressant)
```

Comment:
this should be done once when the drug enters the body and when it leaves the body, instead of on metabolize with a cooldown.